warehouse/ingest-core/src/test/java/datawave/ingest/table/aggregator/GlobalIndexUidAggregatorExpandedTest.java

```diff
@@ -0,0 +1,719 @@
+    }
+    
+    @Test
+    public void exceedMaxAddTwoDropOne() {
+        // Will flipping to seen ignore take into account the available information?
+        // Note this tests both orderings of values.
+        // i.e., count = this.uids.size() - this.uidsToRemove.size();
+        agg = new GlobalIndexUidAggregator(2);
+        Value value1 = UidTestBuilder.newBuilder().withCountOnly(3).build();
+        
+        Value value2 = UidTestBuilder.newBuilder().withUids("uid3", "uid4").build();
+        
+        Value value3 = UidTestBuilder.newBuilder().withUids().withRemovals("uid5").build();
+        
+        Uid.List expectation = UidTestBuilder.valueToUidList(UidTestBuilder.newBuilder().withCountOnly(4).build());
+        
+        testCombinations(expectation, asList(value1, value2, value3));
+    }
+    
+    @Test
+    public void removalToZeroCount() {
+        // Will flipping to seen ignore take into account the available information?
+        // Note this tests both orderings of values.
+        // i.e., count = this.uids.size() - this.uidsToRemove.size();
+        agg = new GlobalIndexUidAggregator(10);
+        Value value1 = UidTestBuilder.newBuilder().withCountOnly(1).build();
+        
+        Value value2 = UidTestBuilder.newBuilder().withUids().withRemovals("uid1").build();
+        
+        Uid.List expectation = UidTestBuilder.valueToUidList(UidTestBuilder.newBuilder().withCountOnly(0).build());
+        
+        testCombinations(expectation, asList(value1, value2));
+    }
+    
+    @Test
+    public void countOnlyReduced() {
+        // Will flipping to seen ignore take into account the available information?
+        // Note this tests both orderings of values.
+        // i.e., count = this.uids.size() - this.uidsToRemove.size();
+        agg = new GlobalIndexUidAggregator(10);
+        Value value1 = UidTestBuilder.newBuilder().withCountOnly(30).build();
+        
+        Value value2 = UidTestBuilder.newBuilder().withUids().withRemovals("uid1", "uid2").build();
+        
+        Uid.List expectation = UidTestBuilder.valueToUidList(UidTestBuilder.newBuilder().withCountOnly(28).build());
+        
+        testCombinations(expectation, asList(value1, value2));
+    }
+    
+    private void testCombinations(Uid.List expectation, List<Value> input) {
+        // There should be nothing in the removal UID list after a Full Major Compaction
+        
+        Uid.List expectNoRemovals = Uid.List.newBuilder().mergeFrom(expectation).clearREMOVEDUID().build();
+        
+        if (!isFullCompactionOnlyTest) {
+            verify("Forward, Partial Major", expectation, testAsPartialCompaction(input));
+        }
+        
+        if (!isPartialCompactionOnlyTest) {
+            verify("Forward, Full Major", expectNoRemovals, testAsFullMajorCompaction(input));
+        }
+        
+        if (!isForwardOnlyTest) {
+            // reverse the ordering of the input and try again
+            Collections.reverse(input);
```

Comment:
I wonder if it's worth clarifying somewhere what forward means in this context. I think the way you've done it is readable for the tests and I wouldn't suggest changing anything, but it's actually opposite the way it happens for real. Keys inserted after are seen first at aggregation time (at least when sensible timestamps are used). So the tests are viewing the list from an aggregation order point of view, not a key insertion point of view. Anyway, I'm not suggesting a change other than possibly documenting somewhere that the tests are looking at things in aggregation order, not key insertion order.

Reply:
Good point.  I added a comment to explain what forward and reverse mean for this test and what they do not mean.  I also added a comment to explain why partial compaction and full compaction were used.